e2e_test/streaming/gap_fill.slt

```diff
@@ -0,0 +1,76 @@
+# Test for GAP FILL in normal streaming mode (without EOWC).
+# This tests the StreamGapFill executor.
+
+statement ok
+CREATE TABLE gap_normal_table (
+    device_id VARCHAR,
+    ts TIMESTAMP,
+    value DOUBLE,
+    PRIMARY KEY (device_id, ts)
+);
+
+statement ok
+INSERT INTO gap_normal_table VALUES
+    ('device1', '2024-05-21 10:00:00', 10.0),
+    ('device1', '2024-05-21 10:02:00', 12.0),
+    ('device1', '2024-05-21 10:05:00', 15.0);
+
+# Use normal streaming syntax (without EMIT ON WINDOW CLOSE)
+statement ok
+CREATE MATERIALIZED VIEW gap_filled_normal AS
+SELECT device_id, ts, value
+FROM GAP_FILL(gap_normal_table, ts, INTERVAL '1' MINUTE);
+
+# Insert some more data
+statement ok
+INSERT INTO gap_normal_table VALUES
+    ('device1', '2024-05-21 10:08:00', 20.0);
```

Comment:
We need to test more cases in the SLT tests, insert in between a gap fill span, insert before.